CRAN (R Programming Language) warning about pugixml.cpp

The R checks for the 'redatam' package that vendors 'pugixml' show this for the released 'pugixml' 1.15

https://cran.r-project.org/web/checks/check_results_redatam.html

```
Check Details

Version: 2.3.0
Check: whether package can be installed
Result: WARN 
  Found the following significant warnings:
    vendor/pugixml.cpp:491:66: warning: ‘*other.pugi::impl::{anonymous}::xml_document_struct::pugi::xml_node_struct.pugi::xml_node_struct::header’ is used uninitialized [-Wuninitialized]
  See ‘/home/hornik/tmp/R.check/r-devel-gcc/Work/PKGS/redatam.Rcheck/00install.out’ for details.
  * used C++ compiler: ‘g++-16 (Debian 16.1.0-1) 16.1.0’
Flavor: [r-devel-linux-x86_64-debian-gcc](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-gcc/redatam-00check.html)

Version: 2.3.0
Check: whether package can be installed
Result: WARN 
  Found the following significant warnings:
    vendor/pugixml.cpp:491:66: warning: ‘*other.pugi::impl::{anonymous}::xml_document_struct::pugi::xml_node_struct.pugi::xml_node_struct::header’ is used uninitialized [-Wuninitialized]
  See ‘/home/hornik/tmp/R.check/r-release-gcc/Work/PKGS/redatam.Rcheck/00install.out’ for details.
  * used C++ compiler: ‘g++-16 (Debian 16-20260425-1) 16.0.1 20260425 (prerelease) [gcc-16 r16-8812-gd9c07462a22]’
Flavor: [r-release-linux-x86_64](https://www.r-project.org/nosvn/R.check/r-release-linux-x86_64/redatam-00check.html)
```

These checks correspond to the 'rchk' subset. 

Because I maintain the 'redatam' package, I proposed a change in a new PR.

Ps. 'pugixml' has been an amazing tool to process Redatam files that have an XML component.

I can reproduce this on GCC 16.0.1, looks like a regression from GCC 15. Full build error:

```
$ g++-16 --version
g++-16 (Ubuntu 16-20260322-1ubuntu1) 16.0.1 20260322 (experimental) [trunk r16-8246-g569ace1fa50]
Copyright (C) 2026 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ g++-16 -O2 -Wall -c src/pugixml.cpp
src/pugixml.cpp: In member function ‘void pugi::xml_document::_move(pugi::xml_document&)’:
src/pugixml.cpp:501:66: warning: ‘*other.pugi::impl::{anonymous}::xml_document_struct::pugi::xml_node_struct.pugi::xml_node_struct::header’ may be used uninitialized [-Wmaybe-uninitialized]
  501 |         #define PUGI_IMPL_GETPAGE(n) PUGI_IMPL_GETPAGE_IMPL((n)->header)
      |                                                             ~~~~~^~~~~~
src/pugixml.cpp:498:169: note: in definition of macro ‘PUGI_IMPL_GETPAGE_IMPL’
  498 |       #define PUGI_IMPL_GETPAGE_IMPL(header) static_cast<impl::xml_memory_page*>(const_cast<void*>(static_cast<const void*>(reinterpret_cast<const char*>(&header) - (header >> 8))))
      |                                                                                                                                                                       ^~~~~~
src/pugixml.cpp:7777:55: note: in expansion of macro ‘PUGI_IMPL_GETPAGE’
 7777 |                 new (other) impl::xml_document_struct(PUGI_IMPL_GETPAGE(other));
      |                                                       ^~~~~~~~~~~~~~~~~
```

--- a/src/pugixml.cpp
+++ b/src/pugixml.cpp
@@ -7774,7 +7774,7 @@ namespace pugi
 		}
 
 		// reset other document
-		new (other) impl::xml_document_struct(PUGI_IMPL_GETPAGE(other));
+		new (other) impl::xml_document_struct(other_page);
 		rhs._buffer = NULL;
 	}
 #endif

